From 8dabf40437ffd7196804ddc8612586c378211acd Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Mon, 10 Jul 2023 17:28:48 +0300
Subject: [PATCH 3/5] DONOTMERGE: Amlogic-specific changes

Changes are not compatible with the remaining platforms.
Keep it as a local patch for now,

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 platform/uboot/bootscript.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/platform/uboot/bootscript.cpp b/platform/uboot/bootscript.cpp
index 35f3367..6158b6b 100644
--- a/platform/uboot/bootscript.cpp
+++ b/platform/uboot/bootscript.cpp
@@ -27,7 +27,7 @@ BOOTLOADER_PARTITION_OVERRIDE()
 #else
 EXTENV(partitions, ";name=bootloader,start=128K,size=130944K,uuid=\${uuid_gpt_bootloader}")
 #endif
-EXTENV(partitions, ";name=uboot-env,size=512K,uuid=\${uuid_gpt_reserved}")
+EXTENV(partitions, ";name=uboot-env,start=3M,size=512K,uuid=\${uuid_gpt_reserved}")
 EXTENV(partitions, ";name=misc,size=512K,uuid=\${uuid_gpt_misc}")
 EXTENV(partitions, ";name=frp,size=512K,uuid=\${uuid_gpt_frp}")
 EXTENV(partitions, ";name=boot_a,size=64M,uuid=\${uuid_gpt_boot_a}")
@@ -47,7 +47,7 @@ EXTENV(partitions, ";name=userdata,size=-,uuid=\${uuid_gpt_userdata}")
 setenv bootargs " init=/init rootwait ro androidboot.boottime=223.708 androidboot.selinux=permissive"
 EXTENV(bootargs, " androidboot.revision=1.0 androidboot.board_id=0x1234567 androidboot.serialno=${serial#}")
 EXTENV(bootargs, " firmware_class.path=/vendor/etc/firmware")
-EXTENV(bootargs, " ${debug_bootargs} printk.devkmsg=on")
+EXTENV(bootargs, " ${debug_bootargs} printk.devkmsg=on gpt")
 
 FUNC_BEGIN(enter_fastboot)
 #ifdef PRE_ENTER_FASTBOOT
@@ -109,7 +109,7 @@ FUNC_END()
 
 FUNC_BEGIN(bootcmd_prepare_env)
  setenv bootdevice_path STRESC(__SYSFS_MMC0_PATH__);
- if test STRESC(${mmc_bootdev}) = STRESC(1);
+ if test STRESC(${mmc_bootdev}) = STRESC(2);
  then
   setenv bootdevice_path STRESC(__SYSFS_MMC1_PATH__);
  fi;
-- 
2.39.2

